lms/static/scripts/postmessage_json_rpc/server/server.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * A JSON-RPC-over-postMessage server.
+ *
+ * On creation the server automatically finds and reads its config settings
+ * from a JSON config object in the document. For example:
+ *
+ *     <script type="application/json" class="js-rpc-server-config">
+ *       { allowedOrigins: ["https://hypothes.is"] }
+ *     </script>
+ *
+ * After constructing a server you have to call its register() method to
+ * register remotely callable methods. After a method has been registered the
+ * server will respond to remote requests for that method by caling the method
+ * and sending the method's return value, serialized to a JSON string, back to
+ * the caller over postMessage. For example:
+ *
+ *     const server = new Server();
+ *     server.register('requestConfig', requestConfig);
+ *
+ */
+export default class Server {
+
+  constructor() {
+    const configEl = document.getElementsByClassName('js-rpc-server-config')[0];
+    const configObj = JSON.parse(configEl.textContent);
+
+    // JSON-RPC messages that don't come from one of these allowed window
+    // origins will be ignored.
+    this._allowedOrigins = configObj.allowedOrigins;
+
+    // Add a postMessage event listener so we can recieve JSON-RPC requests.
+    this._boundReceiveMessage = this._receiveMessage.bind(this);
+    window.addEventListener('message', this._boundReceiveMessage);
+
+    // The methods that can be called remotely via this server.
+    this._registeredMethods = {};
+  }
+
+  /**
+   * Register a remotely callable method with this server.
+   */
+  register(name, method) {
+    this._registeredMethods[name] = method;
+  }
+
+  /**
+   * Turn off this Server instance, it will no longer respond to messages.
+   */
+  off() {
```

Comment:
`off` seems like a slightly odd name for this method. `stop` or `stopListening` perhaps might be clearer?